### ISSUE TITLE
Sitewide, Design system, CMS [AXE-CORE, SCREENREADER]: Heading levels SHOULD increase by one - the alert box heading breaks order

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6375]
#  [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Team affected: VSA Public Websites; Project: AMA ph1
```

Also documented in https://github.com/department-of-veterans-affairs/va.gov-team/issues/7708.

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Heading levels **SHOULD** increase by one. The alert box heading is tagged as an `<h3 class="usa-alert-heading" …` but there isn't an `<h2>` before it. This creates a confusing experience for screen reader users.

This issue happens on the following pages:
- https://www.va.gov/decision-reviews/
- https://www.va.gov/disability/file-an-appeal/
- https://www.va.gov/decision-reviews/supplemental-claim/
- https://www.va.gov/decision-reviews/higher-level-review/
- https://www.va.gov/decision-reviews/board-appeal/
- https://www.va.gov/decision-reviews/multiple-party-claims/

<details><summary>Why it Matters</summary>

The underlying purpose of headers is to convey the structure of the page. For sighted users, the same purpose is achieved using different sizes of text. Text size, however, is not helpful for users of screen readers, because a screen reader identifies a header only if it is properly marked-up. When heading elements are applied correctly, the page becomes much easier to navigate for screen reader users and sighted users alike.

In the same way that sighted users can glance at a page and get a sense of its contents, users of screen readers can do the same by navigating through headings. Well written and properly ordered headings can save users, especially those who use screen readers, a lot of time and frustration.

The purpose of headings is to describe the structure of the webpage, not just highlight important text. They should be brief, clear, unique, and marked with h1 through h6 elements applied in hierarchical order. All of these qualities make headings valuable tools for screen reader users. Similar to the way sighted users can glance at a page and get a sense of its contents, screen reader users can navigate through headings. Well written and properly ordered headings can save screen reader time and frustration.

In addition to making the page more accessible, headings have other benefits since search engines use headings when filtering, ordering, and displaying results. Improving the accessibility of your site can also have the effect of making your page more findable.
</details>

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/disability/file-an-appeal/` in browser
2. Start screenreading device 
3. Read the page
4. Verify that it reads an H1 and then an H3
5. Alternately, run an axe scan in the browser to verify the issue

## Possible Fixes (optional)

The alertbox heading **should** be an H2 with the utility class styling for an H3. 

This is due to the CMS implementing the [VA design system alertbox component](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design/components/alertbox/) as-is. For React apps, there is a `level` prop that takes numeric values. For the alertbox to be implemented in the CMS, it needs to be a static implementation that will work no matter where the alertbox is placed on a page. The heading **should** be coded as:

```
  <h2 class="vads-u-font-size--h3">Alertbox heading</h2>
```

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Heading levels should only increase by one](https://dequeuniversity.com/rules/axe/3.4/heading-order)
* [MDN dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
* [MDN ARIA: alert role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role)
* [MDN Description List element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)

## Screenshots or Trace Logs

![vsa-pw-heading-level-alert](https://user-images.githubusercontent.com/57469/75710161-01005100-5c92-11ea-9d4d-6aeb92677ab0.png)

**Comment by RLHecht** [comment]
@MickinSahni I think this is another ticket that @ncksllvn or @kelsonic would need to address and not me.

**Comment by MickinSahni** [comment]
Hey @jenstrickland 
- can you clarify if this is a Must vs. a Should? I see both in the first comment.

**Comment by ncksllvn** [comment]
This isn't the best fix, but those pages are in Markdown over at vagov-content, so we should just fix the heading tags to be semantic. IDK if this problem is also seen on Drupal pages.

> Consider changing the markup of alert boxes to a description list - <dl> - so that an alert box will not break the heading order of the page. Depending on the purpose of the alert box, it may be helpful to add role="alert" , wrap in a <dialog> , or use some other ARIA to provide context or appropriate notification.

I like this idea, but it sounds like that entails restructuring the markup of a design system [component](https://design.va.gov/components/alertboxes), so it's a big lift and a site-wide issue.

**Comment by jenstrickland** [comment]
@MickinSahni This is a **Should** (just changed the heading to not be a Launchblocker, since the rubric change). I can't find the Must.

**Comment by jenstrickland** [comment]
@ncksllvn @MickinSahni we're discussing with VSP tomorrow about how to handle more global issues, and if there's a short term fix for this particular feature choose that.

**Comment by MickinSahni** [comment]
> @ncksllvn @MickinSahni we're discussing with VSP tomorrow about how to handle more global issues, and if there's a short term fix for this particular feature choose that.

Thanks, Jennifer. Given what we've discussed, I think it makes sense to pass this to Trevor (@1Copenut) for triage. Trevor, can you confirm?

**Comment by 1Copenut** [comment]
@MickinSahni I agree. This one will likely need a refactor on the design system component so devs can modify what heading level is output to the page. I'll add it to the list Jennifer sent me earlier today.

**Comment by jenstrickland** [comment]
@1Copenut - the more I dig into this, a description list seems like it would be really helpful, with the role="alert" or whatever. These need flexibility!

**Comment by MickinSahni** [comment]
Thanks, team. Trevor, I'm removing the `vsa-public-websites` label and the PW Epic related to the AMA changes. Leaving the 508 epic. @jenstrickland @1Copenut

**Comment by Rleahy22** [comment]
PR was merged today and should be on production following the 3:00pm deploy.

The alert box headings should all be `h2`s at the urls Josh mentioned:
https://www.va.gov/decision-reviews/
https://www.va.gov/disability/file-an-appeal/
https://www.va.gov/decision-reviews/supplemental-claim/
https://www.va.gov/decision-reviews/higher-level-review/
https://www.va.gov/decision-reviews/board-appeal/
https://www.va.gov/decision-reviews/multiple-party-claims/

**Comment by brianalloyd** [comment]
Validated, closing in support.
